src/librustdoc/html/templates/sidebar.html

```diff
@@ -0,0 +1,32 @@
+{%- if it.is_struct()
+        || it.is_trait()
+        || it.is_primitive()
+        || it.is_union()
+        || it.is_enum()
+        || it.is_mod()
+        || it.is_typedef() -%}
+<h2 class="location"><a href="#">
+{%- match it.kind.as_ref() -%}
+{%- when clean::ModuleItem with (_) -%}
+{%- if it.is_crate() %}Crate{% else %}Module{% endif %} {#+ -#}
+{%- else -%}
+{%- endmatch -%}
+{{- it.name.as_ref().unwrap() -}}
+</a></h2>{% endif -%}
```

Comment:
I would love to see this simplified as:

```
{%- if !title.is_empty() %-}
<h2 class="location">
  <a href="#">{{ title_prefix }}{{ title}}</a>
</h2>
{%- endif -%}
```

This would necessitate giving `Sidebar` a `title: &str` and `title_prefix: &str` field, and moving the `it.is_foo()` logic, plus the "Crate " / "Module " / "" logic back into the Rust code.